api/src/jwt/jwtPermissionMap.ts

```diff
@@ -0,0 +1,74 @@
+import { Uuid } from "../enums";
+import * as JWT from "jsonwebtoken";
+
+export type PermissionMap = {
+    jwt: Map<string, Map<string, (jwt) => void>>;
+};
+
+export type jwtPermissions = {
+    groups: Array<Uuid>;
+    userId?: Uuid;
+};
+
+/**
+ * Parse the permission map from the PROCESS_MAP environmental variable to an object
+ * @param permissionMap - Configuration instance
+ * @returns - Parsed permissions map
+ */
+export function parsePermissionMap(permissionMap: string): PermissionMap {
+    // Parse permission map
+    try {
+        const map = JSON.parse(permissionMap);
+
+        // Evaluate stringified functions
+        if (map.jwt && map.jwt.groups) {
+            Object.keys(map.jwt.groups).forEach((key) => {
+                map.jwt.groups[key] = eval(map.jwt.groups[key]);
+            });
+        }
+        if (map.jwt && map.jwt.userId) {
+            Object.keys(map.jwt.userId).forEach((key) => {
+                map.jwt.userId[key] = eval(map.jwt.userId[key]);
+            });
+        }
+
+        return map as PermissionMap;
+    } catch (err) {
+        // TODO: Add error logging provider
+        console.log(err.message);
```

Comment:
Is this correct? I see it in some places that we do a generic try-catch, in this case we're not returning a default. Can the code that uses this method continue then, or should this actually throw? Maybe a test is needed for this?

Reply:
Fixed in following push